File: 1099-two-sum-less-than-k/1099-two-sum-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var twoSumLessThanK = function(nums, k) {
    
    nums.sort((a,b) => a - b)
    
    let start = 0
    let end = nums.length - 1
    let max = Number.MIN_SAFE_INTEGER;
    
    while(start < end){
        if (nums[start] + nums[end] < k){
            max = Math.max(max, nums[start] + nums[end])
            start++
        }else{
            end--
        }
    }
    
    return max === Number.MIN_SAFE_INTEGER ? -1 : max;
    
};

File: 1099-two-sum-less-than-k/1099-two-sum-less-than-k.py
class Solution:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        
        nums.sort()
        
        left = 0
        right = len(nums) - 1
        maxSum = 0
        
        while left < right:
            currSum = nums[left] + nums[right]
            if currSum < k:
                if maxSum < currSum:
                    maxSum = currSum
                left += 1
            else:
                right -= 1
        
        return maxSum if maxSum else -1